src/maliput_malidrive/xodr/lane_road_mark.cc

```diff
@@ -0,0 +1,184 @@
+
+const std::map<LaneRoadMark::LaneChange, std::string> lane_change_to_str_map{
+    {LaneRoadMark::LaneChange::kBoth, "both"},
+    {LaneRoadMark::LaneChange::kDecrease, "decrease"},
+    {LaneRoadMark::LaneChange::kIncrease, "increase"},
+    {LaneRoadMark::LaneChange::kNone, "none"},
+};
+
+const std::map<std::string, LaneRoadMark::LaneChange> str_to_lane_change_map{
+    {"both", LaneRoadMark::LaneChange::kBoth},
+    {"decrease", LaneRoadMark::LaneChange::kDecrease},
+    {"increase", LaneRoadMark::LaneChange::kIncrease},
+    {"none", LaneRoadMark::LaneChange::kNone},
+};
+
+const std::map<std::string, LaneRoadMark::Type> str_to_type_map{
+    {"botts dots", LaneRoadMark::Type::kBottsDots},
+    {"broken broken", LaneRoadMark::Type::kBrokenBroken},
+    {"broken solid", LaneRoadMark::Type::kBrokenSolid},
+    {"broken", LaneRoadMark::Type::kBroken},
+    {"curb", LaneRoadMark::Type::kCurb},
+    {"custom", LaneRoadMark::Type::kCustom},
+    {"edge", LaneRoadMark::Type::kEdge},
+    {"grass", LaneRoadMark::Type::kGrass},
+    {"none", LaneRoadMark::Type::kNone},
+    {"solid broken", LaneRoadMark::Type::kSolidBroken},
+    {"solid solid", LaneRoadMark::Type::kSolidSolid},
+    {"solid", LaneRoadMark::Type::kSolid},
+};
+
+const std::map<LaneRoadMark::Type, std::string> type_to_str_map{
+    {LaneRoadMark::Type::kBottsDots, "botts dots"},
+    {LaneRoadMark::Type::kBrokenBroken, "broken broken"},
+    {LaneRoadMark::Type::kBrokenSolid, "broken solid"},
+    {LaneRoadMark::Type::kBroken, "broken"},
+    {LaneRoadMark::Type::kCurb, "curb"},
+    {LaneRoadMark::Type::kCustom, "custom"},
+    {LaneRoadMark::Type::kEdge, "edge"},
+    {LaneRoadMark::Type::kGrass, "grass"},
+    {LaneRoadMark::Type::kNone, "none"},
+    {LaneRoadMark::Type::kSolidBroken, "solid broken"},
+    {LaneRoadMark::Type::kSolidSolid, "solid solid"},
+    {LaneRoadMark::Type::kSolid, "solid"},
+};
+
+const std::map<LaneRoadMark::Weight, std::string> weight_to_str_map{
+    {LaneRoadMark::Weight::kBold, "bold"},
+    {LaneRoadMark::Weight::kStandard, "standard"},
+};
+
+const std::map<std::string, LaneRoadMark::Weight> str_to_weight_map{
+    {"bold", LaneRoadMark::Weight::kBold},
+    {"standard", LaneRoadMark::Weight::kStandard},
+};
+
+const std::map<Rule, std::string> rule_to_str_map{
+    {Rule::kCaution, "caution"},
+    {Rule::kNoPassing, "no passing"},
+    {Rule::kNone, "none"},
+};
+
+const std::map<std::string, Rule> str_to_rule_map{
+    {"caution", Rule::kCaution},
+    {"no passing", Rule::kNoPassing},
+    {"none", Rule::kNone},
+};
+
+}  // namespace
+
+/// Roadmark
+
+std::string LaneRoadMark::lane_change_to_str(LaneChange lane_change) { return lane_change_to_str_map.at(lane_change); }
+
+LaneRoadMark::LaneChange LaneRoadMark::str_to_lane_change(const std::string& lane_change) {
+  if (str_to_lane_change_map.find(lane_change) == str_to_lane_change_map.end()) {
+    MALIDRIVE_THROW_MESSAGE(lane_change + " lane road mark lane change is not available.");
```

Comment:
Consider throwing a specific error type from `maliput::common::error`. In this case it'd make sense to throw a `road_network_description_parser_error`, which is "API" for `xodr_parsing_error`.